vertx-mysql-client/src/test/java/io/vertx/mysqlclient/MySQLTLSTest.java

```diff
@@ -191,4 +192,29 @@ public void testConnFail(TestContext ctx) {
     MySQLConnection.connect(vertx, options, ctx.asyncAssertFailure(error -> {
     }));
   }
+
+  @Test
+  public void testChangeUser(TestContext ctx) {
+    options.setSslMode(SslMode.REQUIRED);
+    options.setPemTrustOptions(new PemTrustOptions().addCertPath("tls/files/ca.pem"));
+
+    MySQLConnection.connect(vertx, options, ctx.asyncAssertSuccess(conn -> {
+      conn.query("SELECT current_user()", ctx.asyncAssertSuccess(res1 -> {
+        Row row1 = res1.iterator().next();
+        String username = row1.getString(0);
+        ctx.assertEquals("mysql", username.substring(0, username.lastIndexOf('@')));
+        MySQLConnectOptions changeUserOptions = new MySQLConnectOptions()
+          .setUser("superuser")
+          .setPassword("password")
+          .setDatabase("emptyschema");
+        conn.changeUser(changeUserOptions, ctx.asyncAssertSuccess(v2 -> {
```

Comment:
why are we reusing `MySQLConnectionOptions` ? does the `changeUser` method needs to perform a reconnection ? if not then we should have a specific options class for this instead of reusing the connect options

Reply:
Change user will perform a re-authentication but not reconnect. The options used here is for reusing a subset of the connect options to authenticate, do you think it’s necessary to define a new class here? I think the code may be duplicated with `SqlConnectOptions` and `MySQLConnectOptions`.

Reply:
yes I think it's better to have a new options class usability wise

Reply:
perhaps also we should offer to do automatic user change at connection time

Reply:
what does `automatic user change at connection time` mean?

Reply:
The `CHANGE_USER` API is updated in https://github.com/eclipse-vertx/vertx-sql-client/pull/420/commits/f083dd128daf3a0772bb7b3672a5648c17b48a8f.